core/jobs/batch_jobs/voiceover_migration_job.py

```diff
@@ -0,0 +1,358 @@
+
+        voice_artist_link = (
+            element[1]['voice_artist_link'][0])
+        assert isinstance(
+            voice_artist_link,
+            voiceover_models.ExplorationVoiceArtistsLinkModel
+        )
+
+        voice_artist_id_to_language_code_mapping = {}
+
+        content_id_to_voiceovers_mapping = (
+            voice_artist_link.content_id_to_voiceovers_mapping)
+
+        for voice_artist_metadata_model in voice_artist_metadata_models_list:
+            voice_artist_id_to_language_code_mapping[
+                voice_artist_metadata_model.id] = (
+                    voice_artist_metadata_model.language_code_to_accent)
+
+        entity_id = voice_artist_link.id
+        entity_type = 'exploration'
+        entity_version = exploration_model.version
+
+        entity_voiceover_id_to_entity_voiceovers: Dict[
+            str, voiceover_models.EntityVoiceoversModel] = {}
+
+        for content_id, lang_code_to_voiceover_mapping in (
+                content_id_to_voiceovers_mapping.items()):
+            for lang_code, voiceover_mapping in (
+                    lang_code_to_voiceover_mapping.items()):
+                voice_artist_id = voiceover_mapping[0]
+                voiceover_dict = voiceover_mapping[1]
+
+                try:
+                    accent_code = voice_artist_id_to_language_code_mapping[
+                        voice_artist_id][lang_code]
+                except Exception as e:
+                    raise Exception(
+                        'Please assign all the accents for voice artists in '
+                        'language code %s.' % lang_code) from e
+
+                entity_voiceover_id_to_entity_voiceovers = (
+                    cls.update_entity_voiceover_for_given_id(
+                        entity_type,
+                        entity_id,
+                        entity_version,
+                        accent_code,
+                        content_id,
+                        voiceover_dict,
+                        entity_voiceover_id_to_entity_voiceovers
+                    )
+                )
+        return list(entity_voiceover_id_to_entity_voiceovers.values())
+
+    @classmethod
+    def count_voiceovers(
+        cls,
+        entity_voiceovers_model: voiceover_models.EntityVoiceoversModel
+    ) -> int:
+        """Calculates the number of voiceovers for a given entity
+        voiceover model.
+
+        Args:
+            entity_voiceovers_model: EntityVoiceoversModel. An instance of an
+                entity voiceover model from which the number of voiceovers will
+                be calculated.
+
+        Returns:
+            int. The number of voiceovers present in the given entity
+            voiceovers model.
+        """
+        total_voiceovers: int = 0
+        for voiceover_type_to_voiceovers in (
+                entity_voiceovers_model.voiceovers.values()):
+            if 'manual' in voiceover_type_to_voiceovers:
```

Comment:
The keys of any sub-objects should be well-defined and always the same (for each instance) so that checks like this are not necessary.